src/dynamodb/batch/duplicate-handling/filter.js

```diff
@@ -0,0 +1,102 @@
+function filterUniqueKeys(keys = []) {
+  if (!keys || !keys.length) return [];
+  if (keys.some((key) => typeof key !== 'object'))
+    throw new Error('All keys should must be objects.');
+
+  const stringifiedKeys = [];
+  const uniqueKeys = [];
+
+  keys.forEach((key) => {
+    if (
+      !stringifiedKeys.find((stringKey) => stringKey === JSON.stringify(key))
```

Comment:
Why stringify?

Reply:
But isn't this just the object key?

Reply:
Yes but the key is an object and can consist of multiple attributes

Reply:
I will go through the naming for the key comparison code and see if that can clarify it a bit